errors/errors.go

```diff
@@ -0,0 +1,195 @@
+package errors
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+var (
+	// InternalErr represents a general case issue that cannot be
+	// categorized as any of the below cases.
+	InternalErr = NewRootError(0, "internal")
+
+	// UnauthorizedErr is used whenever a request without sufficient
+	// authorization is handled.
+	UnauthorizedErr = NewRootError(1, "unauthorized")
+
+	// NotFoundErr is used when a requested operation cannot be completed
+	// due to missing data.
+	NotFoundErr = NewRootError(2, "not found")
+
+	// InvalidMsgErr is returned whenever an event is invalid and cannot be
+	// handled.
+	InvalidMsgErr = NewRootError(3, "invalid message")
+
+	// InvalidModelErr is returned whenever a message is invalid and cannot
+	// be used (ie. persisted).
+	InvalidModelErr = NewRootError(4, "invalid model")
+)
+
+// NewRootError returns an error instance that should be used as the base for
+// creating error instances during runtime.
+//
+// Popular root errors are declared in this package, but extensions may want to
+// declare custom codes. This function ensures that no error code is used
+// twice. Attempt to reuse an error code results in panic.
+//
+// Use this function only during a program startup phase.
+func NewRootError(code uint32, description string) RootError {
```

Comment:
pp: this should be renamed to `RegisterErrorCode` or something similar to make it very clear that this function not only creates a new error instance but also keeps an unique index in memory. Maybe it is even better to split it into 2 functions. Otherwise it may be accidentally be called by somebody to just create a new error instance at runtime. It should be used with an `var ErrXXXX` initialization only, so that we can have it fail fast.

Reply:
I agree. @ruseinov was also suggesting changing this name. Let's think of a good name for both the function and the error structure, that will represent the intentions well. :+1: 

I think `RegisterErrorCode` is not good enough, because it's not only registering the error code. It is creating an error as well, which in my opinion is more important than the error code.
While thinking of it, I was imagining that errors created with this function are the top level errors that can be either used as they are or inherited (wrapped) to create child errors. 

Please just write down proposals and explanation if necessary.

Reply:
How about one of these:
```
ConflictErr = errors.Register( errors.New(4, "not unique"))
```
```
ConflictErr = errors.RegisterNew(4, "not unique"))
```

Reply:
I like the second one :+1: 

First one requires `New` function, which right now exists with a different notation. I think having `New` function might make one think that it is to be used during the runtime as well.

Reply:
q: if you like the second option, too can we rename the function before merging this?

Reply:
@husio @alpe why do we need RegisterNew? Can it just be register? I'm not following

Reply:
This set of comments is about renaming `NewRootError` function. It might make sense to rename `RootError` as well, which you mentioned earlier.

Reply:
I underatand.
Why don’t we use Error instead of RootError and Register instead of NewRootError?

Reply:
👍 to rename `RootError`